src/openforms/prefill/service.py

```diff
@@ -0,0 +1,188 @@
+
+    For each submission value variable that need to be prefilled, the according plugin will
+    be used to fetch the value. If ``register`` is not specified, the default registry instance
+    will be used.
+    """
+    from openforms.formio.service import normalize_value_for_component
+
+    from .registry import register as default_register
+
+    register = register or default_register
+
+    state = submission.load_submission_value_variables_state()
+    variables_to_prefill = state.get_prefill_variables()
+
+    key_source_mappings: defaultdict[str, str] = defaultdict()
+    if state.variables:
+        for variable_key, variable in state.variables.items():
+            key_source_mappings[variable_key] = variable.form_variable.source
+
+    variables_with_attribute: list[SubmissionValueVariable] = []
+    variables_with_options: list[SubmissionValueVariable] = []
+    prefill_data: defaultdict[str, JSONEncodable] = defaultdict(dict)
+
+    for variable in variables_to_prefill:
+        key_source_mappings[variable.form_variable.key] = variable.form_variable.source
+        # variables which have prefill enabled via the component's configuration
+        if (
+            variable.form_variable.source == FormVariableSources.component
+            and variable.form_variable.prefill_plugin
+            and variable.form_variable.prefill_attribute
+        ):
+            variables_with_attribute.append(variable)
+
+        if variable.form_variable.source == FormVariableSources.user_defined:
+            # variables which have prefill enabled via the variables tab and define prefill options
+            if (
+                variable.form_variable.prefill_plugin
+                and variable.form_variable.prefill_options
+            ):
+                variables_with_options.append(variable)
+
+            # variables which have prefill enabled via the variables tab and define prefill attribute
+            if (
+                variable.form_variable.prefill_plugin
+                and variable.form_variable.prefill_attribute
+            ):
+                variables_with_attribute.append(variable)
+
+    if variables_with_attribute and (
+        results_from_attribute := fetch_prefill_values_from_attribute(
+            submission, register, variables_with_attribute
+        )
+    ):
+        prefill_data.update(**results_from_attribute)
+    if variables_with_options and (
+        results_from_options := fetch_prefill_values_from_options(
+            submission, register, variables_with_options
+        )
+    ):
+        prefill_data.update(**results_from_options)
+
+    total_config_wrapper = submission.total_configuration_wrapper
+    for variable_key, prefill_value in prefill_data.items():
+        if key_source_mappings[variable_key] == FormVariableSources.component:
+            try:
+                component = total_config_wrapper.component_map[variable_key]
+            except KeyError:
+                return None
```

Comment:
This looks a bit weird - if something is wrong we seem to discard everything. Does that make sense? The lack of coverage on the error handling also makes this seem like a what-if.

This could occur if someone manually creates a `FormVariable` in the admin/shell without it being backed by an actual form definition component, which shouldn't happen. The API endpoint synchronizes the components and derived variables, so I'm leaning towards removing the try-except and if it crashes, it crashes and at least it will become visible in Sentry instead of failing silently, no?

In the old situation this wouldn't happen anyway because we looped only over the prefill variables.